### DISCUSSION TITLE
Possible to sumbit in child?

### DISCUSSION THREAD

**Discussion by benesva4** [react-hook-form/react-hook-form#3893] [Q&A] [answered]
Hello, Is it possible to somehow associate submit function with a form and then call it in a child? My form is long, so it's made of a few components, that use FormContext, but when I want to submit from those child components I need to create new submit function in all of them. Is there a way around it? 

Simple example here: https://codesandbox.io/s/rhf-child-submit-7ii4q

I know that i can add onBlur and onChange on <form /> itself, but It's not good solution especially if you need to submit here on sumbit and there onChange. I can pass the submit function via props or via context, but it seems a little burdensome. I can even hijack <FormContext /> as I did and commented out in the example, but it seems like a terrible idea, especially in a typescript environment. 

To me it seems like there should be something like: 
`const { submit } => useForm({ onSubmit: (values) => { ... } })`

Thank you very much for help :-)

**Comment by bluebill1049** [comment, marked as answer]
`onClick={() => handleSubmit(() => { console.log('submit') })()}`?

**Reply by benesva4** [reply]
I don't understand. This way I have to create a submit handler in every child component I want to use it, without using a submit button, right?

**Reply by bluebill1049** [reply]
if you are using context, you can use `handleSubmit` in your child component right?

**Reply by benesva4** [reply]
I don't need to share handleSubmit, but the function I put inside that handleSubmit (console.log('submit') in your example). Yes I could share that in many not so comfortable ways. I just wanted to know, if it's possible to do it through rhf somehow. Because if the submit handler is complex, then I have to write that handleSubmit(() => { console.log('submit') })()} multiple times. Like in that codebox I sent, where there is something added before submitting. Thanks for the insane response times by the way :D

**Reply by bluebill1049** [reply]
no worries, we don't have the API that you described above (but will keep a note about it).

**Reply by benesva4** [reply]
OK, thank you very mcuh. I'll share it by some other mean. Have a nice day and keep up the great work.
